org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/model/ContentModelProvider.java

```diff
@@ -63,19 +63,58 @@ public String getKind() {
 	 * (xsi:schemaLocation, xsi:noNamespaceSchemaLocation, doctype) and null
 	 * otherwise.
 	 * 
-	 * @param document
-	 * @param internal
+	 * @param document the DOM document.
+	 * @param internal true if it is an internal content model (ex : DOCCTYPE
+	 *                 subset) and false otherwise.
+	 * 
 	 * @return the content model provider by using standard association
 	 *         (xsi:schemaLocation, xsi:noNamespaceSchemaLocation, doctype) and null
 	 *         otherwise.
 	 */
 	boolean adaptFor(DOMDocument document, boolean internal);
 
+	/**
+	 * Returns true if the given resource uri can be adapted for this content model
+	 * and false otherwise.
+	 * 
+	 * o@param uri the resource Uri.
+	 * 
+	 * @return true if the given resource uri can be adapted for this content model
+	 *         and false otherwise.
+	 */
 	boolean adaptFor(String uri);
 
+	/**
+	 * Returns the identifiers list from the given document and namespace.
+	 * 
+	 * @param xmlDocument  the DOM document.
+	 * @param namespaceURI the namespace.
+	 * 
+	 * @return the identifiers list from the given document and namespace.
+	 */
 	Collection<Identifier> getIdentifiers(DOMDocument xmlDocument, String namespaceURI);
 
-	CMDocument createCMDocument(String key);
+	/**
+	 * Create content model document (XSD, DTD, etc) from the given resource key and
+	 * null otherwise.
+	 * 
+	 * @param key                     the resource key.
+	 * @param resolveExternalEntities true if external entities can be resolved and
+	 *                                false otherwise.
+	 * @return the content model document (XSD, DTD, etc) from the given resource
+	 *         key and null otherwise.
+	 */
+	CMDocument createCMDocument(String key, boolean resolveExternalEntities);

```

Comment:
resolveExternalEntities or downloadExternalResources?

Reply:
- resolveExternalEntities is used to configure the Xerces SAX parser when any XML, DTD, XSD is loaded. The resolveExternalEntities is a feature that Xerces supports.
 - downloadRemoteResources is managed on CacheResourcesManager side when an URL must be downloaded. Its XMLCacheResolverExtension an implementation of Xerces XMLEntityResolver which provides the capability to returns the content of an URI.

In other words, we need only resolveExternalEntities to configure Xerces SAX parser.